content/en/tracing/connect_logs_and_traces/java.md

```diff
@@ -17,25 +17,31 @@ further_reading:
       text: 'Correlate request logs with traces automatically'
 ---
 
-## Automatic Trace ID injection
+## Automatic Injection into Logs
 
-Enable injection in the Java tracer's [configuration][1] by adding `-Ddd.logs.injection=true` as a jvm startup argument or through environment variable `DD_LOGS_INJECTION=true`.
+Enable injection in the Java tracer's [configuration][1] through the environment variable `DD_LOGS_INJECTION=true` or the JVM startup argument `-Ddd.logs.injection=true`.
+If you haven't done so already, we recommend configuring the Java tracer with `DD_ENV`, `DD_SERVICE`, and `DD_VERSION`. This will provide the smoothest
+experience for adding `env`, `service`, and `version` (see [Unified Service Tagging][2] for more details).
 
-If your logs are raw formatted, update your formatter to include `dd.trace_id` and `dd.span_id` in your logger configuration:
+If your logs are raw formatted, update your formatter to include `dd.env`, `dd.service`, `dd.version`, `dd.trace_id`, and `dd.span_id` in your logger configuration:
 
 ```xml
-<Pattern>"%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %X{dd.trace_id:-0} %X{dd.span_id:-0} - %m%n"</Pattern>
+<Pattern>"%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %X{dd.env} %X{dd.service} %X{dd.version} %X{dd.trace_id:-0} %X{dd.span_id:-0} - %m%n"</Pattern>
```

Comment:
@PierreVanRoy let's update this section and refer to it from the Log java collection document.

That will avoid maintaining it in several places.